src/components/Tool/Appeal/Shared/useGetPledgeOrDonation/useGetPledgeOrDonation.ts

```diff
@@ -15,7 +13,7 @@ type FormatPledgeOrDonationProps = {
   appealStatus: AppealStatusEnum;
   dateOrFrequency?: PledgeFrequencyEnum | string | null;
   locale: string;
-  t: TFunction;
+  getLocalizedPledgeFrequency: (freqEnum: string | null | undefined) => string;
```

Comment:
You could make the function into a type, which would make it easier to maintain. - src/hooks/useLocalizedConstants.ts